### PR TITLE
Gateway recovery

### PR DESCRIPTION
**Pull request by piyushk**
This PR can be checked in after the demo is over. I don't think there is any rush to commit it in right now.

@stonier: This might require a thorough code review

### CODE REVIEW

**Comment by piyushk** [comment]
Also note that this PR does not solve synchronization issues between 2 gateways. I would like to deal with that in #213

File: rocon_gateway/src/rocon_gateway/gateway_hub.py [hunk 1]
```diff
@@ -19,10 +19,32 @@
...
+##############################################################################
+
+class HubConnectionCheckerThread(threading.Thread):
+    '''
+      Pings redis periodically to figure out if redis is still alive.
+    '''
+    def __init__(self, ip, port, hub_connection_lost_hook):
+        threading.Thread.__init__(self)
+        self.daemon = True # clean shut down of thread when hub connection is lost
+        self.ping_frequency = 0.2 # Too spammy? # TODO Need to parametrize
```
Comment (stonier) [inline rocon_gateway/src/rocon_gateway/gateway_hub.py:36]: I'm fine with leaving this hardcoded until it either us annoys us while testing or users actually have a use case where it needs to be varied from robot to robot.

**Comment by stonier** [comment]
This looks good, and certainly doesn't interfere with anything we're running, so I'll merge it. I've only got one question from me though, which we can discuss in #219.

**Comment by stonier** [comment]
Another thing I observed - I tested with local_wireless.launch (hub, gateway, cpp talker) and remote_wireless.launch (gateway, cpp listener). 

When I dropped the wireless networking on the remote system, and then reconnected within a short period (~30 seconds), talker and listener got chatty again.

If I left it for some time before restarting, roscpp threw the following error:

```
[ERROR] [1385625801.225458594]: couldn't resolve publisher host [snorriwork.yujin.com]
```

which I am guessing totally breaks the tcp/ip connection between the two and it can't reconnect, even after restarting the network.

Not sure what the behaviour of the unreliable transports are yet - curious.

Is this irrelevant I wonder? Synchronization issues can be discussed further in #213.

**Comment by piyushk** [comment]
I'll make a note to test synchronization after this method has been displayed. I have no idea how roscpp/rospy behave once the tcp communication breaks down.

**Comment by stonier** [comment]
Location of that error above in the sources: https://github.com/ros/ros_comm/blob/hydro-devel/clients/roscpp/src/libros/transport/transport_tcp.cpp#L242

**Pull request merged**